src/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -145,6 +145,40 @@ function commandHasRequiredField(command)
   return command.arguments.some(arg => !arg.isOptional);
 }
 
+function enumName(clusterName, enumLabel)
+{
+  clusterName = appHelper.asUpperCamelCase(clusterName);
+  enumLabel   = appHelper.asUpperCamelCase(enumLabel);
+  // Some enum names have one or more copies of the cluster name at the
+  // beginning.
+  while (enumLabel.startsWith(clusterName)) {
+    enumLabel = enumLabel.replace(new RegExp("^" + clusterName), "");
+  }
+
+  if (enumLabel.endsWith("Enum")) {
+    // Strip that off; it'll clearly be an enum anyway.
+    enumLabel = enumLabel.replace(/Enum$/, "");
```

Comment:
same here, maybe some `enumLabel.substring(0, enumLabel.length - 4)` works.

Reply:
Yes, that should be fine.